gap/Forms.gi

```diff
@@ -1,56 +1,77 @@
 # generators used during construction of the group immutable matrices over the
 # appropriate field.
 InstallGlobalFunction("ConjugateToSesquilinearForm",
 function(group, type, gramMatrix)
     local gapForm, newForm, gapToCanonical, canonicalToNew, field, formMatrix,
-        result;
+        result, d;
     if not type in ["S", "O", "U"] then
         ErrorNoReturn("<type> must be one of 'S', 'U', 'O', but <type> = ",
                       type);
     fi;
+    d := DimensionOfMatrixGroup(group);
     field := DefaultFieldOfMatrixGroup(group);
     if type = "S" or type = "O" then
-        gapForm := BilinearFormByMatrix(InvariantBilinearForm(group).matrix, 
-                                        field);
+        formMatrix := BilinearForm(group, type);
+        if formMatrix = fail then
+            if type = "S" then
+                ErrorNoReturn("No preserved symplectic form found for <group>");
+            else
+                ErrorNoReturn("No preserved symmetric bilinear form found for", 
+                              " <group>");
+            fi;
+        fi;
+        gapForm := BilinearFormByMatrix(formMatrix, field);
         newForm := BilinearFormByMatrix(gramMatrix, field);
     else
-        if HasInvariantSesquilinearForm(group) then
-            gapForm := HermitianFormByMatrix(InvariantSesquilinearForm(group).matrix,
-                                             field);
-        else    
-            formMatrix := UnitaryForm(group);
-            if formMatrix = fail then
-                Error("No preserved unitary form found for <group>");
-            fi;
-            gapForm := HermitianFormByMatrix(formMatrix, field);
+        formMatrix := UnitaryForm(group);
+        if formMatrix = fail then
+            ErrorNoReturn("No preserved unitary form found for <group>");
         fi;
+        gapForm := HermitianFormByMatrix(formMatrix, field);
         newForm := HermitianFormByMatrix(gramMatrix, field);
     fi;
-    # the following if condition can only ever be fulfilled if <group> is an
-    # orthogonal group; there the case of even dimension is problematic since,
-    # in that case, there are two similarity classes of bilinear forms
-    if not WittIndex(gapForm) = WittIndex(newForm) then
-       ErrorNoReturn("The form preserved by <group> must be similar to the form ", 
-                     "described by the Gram matrix <gramMatrix>.");
-    fi;
-    gapToCanonical := BaseChangeHomomorphism(BaseChangeToCanonical(gapForm), 
-                                             field);
-    canonicalToNew := BaseChangeHomomorphism(BaseChangeToCanonical(newForm) ^ (-1), 
-                                             field);
-    result := MatrixGroup(field, canonicalToNew(gapToCanonical(GeneratorsOfGroup(group))));
-    if HasSize(group) then
-        SetSize(result, Size(group));
+    # The Form package has a bug for d = 1 so we need to make this exception
+    if not d = 1 then
```

Comment:
```suggestion
    if d <> 1 then
```

Reply:
Also, how about handling the case were the forms are identical a bit more efficiently?
```suggestion
    if gapForm = newForm then
      # nothing to be done
      result := group;
    elif d <> 1 then
```